xbmc/cores/RetroPlayer/rendering/RPRenderManager.h

```diff
@@ -205,7 +206,7 @@ class CRPRenderManager : public IRenderManager, public IRenderCallback
   std::set<std::shared_ptr<CRPBaseRenderer>> m_renderers;
   std::vector<IRenderBuffer*> m_pendingBuffers; // Only access from game thread
   std::vector<IRenderBuffer*> m_renderBuffers;
-  std::map<AVPixelFormat, SwsContext*> m_scalers;
+  std::map<AVPixelFormat, std::map<AVPixelFormat, SwsContext*>> m_scalers; // From -> to -> context
```

Comment:
Might be a nice opportunity for a unique_ptr with a custom deleted for SwsContext.

Reply:
I think we have an avalanche of code to review for the savestate manager and I'd rather leave incremental improvements out and keep it to fixes and features.